src/query/utils.rs

```diff
@@ -41,6 +44,55 @@ where
         .and_then(|transport_cell| from_connection(&transport_cell, compression))
 }
 
+pub fn send_frame_to_all_connections<S, T, M>(sender: &S, frame_bytes: Vec<u8>) -> error::Result<Frame>
+    where
+        S: GetAllConnections<T, M> + GetCompressor<'static> + Sized,
+        T: CDRSTransport + 'static,
+        M: r2d2::ManageConnection<Connection=RefCell<T>, Error=error::Error> + Sized,
+{
+    let ref compression = sender.get_compressor();
+
+    let mut results: Vec<error::Result<Frame>> = Vec::new();
+
+    for connection in sender.get_all_connections() {
+        match connection {
+            Some(conn) => {
+                results.push(write_frame_to_connection(&frame_bytes, compression, conn));
+            }
+            None => { results.push(transform_error()); }
+        }
+    }
+
+    let result = results.iter().find(|r| r.is_err())
+        .unwrap_or_else( ||get_any_valid_result(&results));
```

Comment:
Unfortunately, getting any valid result may not work as expected. Each node will prepare a query with a different ID. That's why if we have a cluster with >1 nodes load balancer may select a "wrong" node which has the query prepared with a different ID.

What I had been thinking about is to have a map of prepared queries to nodes so that during preparing new entity is added to this map. Whenever `exec`-like call is done, the executor picks up a node where the prepared query ID is reside. However for the a `LoadBalancingStrategy` should be able to return a node basing on providing criteria, e.g.

```rust
load_balancer
  .find_node(|node_in_cluster| node.addr() == addr_assigned_to_prepared_id)
  .send_frame(frame_bytes)
```

Reply:
Good idea, I will work on it.